controllers/servicefence_controller.go

```diff
@@ -317,21 +326,123 @@ func (r *ServicefenceReconciler) updateVisitedHostStatus(host *lazyloadv1alpha1.
 	domains := make(map[string]*lazyloadv1alpha1.Destinations)
 	now := time.Now().Unix()
 
-	// update status of destinations of hosts with config
+	// update domains with spec.host
 	for h, strategy := range host.Spec.Host {
-		allHost := []string{h}
-		if hs := getDestination(h); len(hs) > 0 {
+
+		fullHost := h
+		subDomains := strings.Split(h, ".")
+		switch len(subDomains) {
+		case 5:
+		case 1:
+			fullHost = fmt.Sprintf("%s.%s.svc.cluster.local", subDomains[0], host.Namespace)
+		case 2:
+			fullHost = fmt.Sprintf("%s.%s.svc.cluster.local", subDomains[0], subDomains[1])
+		default:
+			log.Errorf("%s is invalid host, skip", h)
+			continue
+		}
+		if !isValidHost(fullHost) {
+			continue
+		}
+		if domains[fullHost] != nil {
+			continue
+		}
+
+		allHost := []string{fullHost}
+		if hs := getDestination(fullHost); len(hs) > 0 {
 			allHost = append(allHost, hs...)
 		}
 
-		domains[h] = &lazyloadv1alpha1.Destinations{
+		domains[fullHost] = &lazyloadv1alpha1.Destinations{
 			Hosts:  allHost,
 			Status: checkStatus(now, strategy),
 		}
 	}
 
-	// update status with metric status
-	// -> (visited)host -> destinations
+	// update domains with spec.namespace
+	for _, ns := range host.Spec.NamespaceSelector {
+		svcs, err := r.env.K8SClient.CoreV1().Services(ns).List(metav1.ListOptions{})
```

Comment:
why we need to list service from k8s when we already have svcCache.

Reply:
add nsSvcCache in memory now